.github/workflows/release.yml

```diff
@@ -308,26 +308,31 @@ jobs:
           fi
           echo "TAG: $TAG"
           echo "TAG=$TAG" >> $GITHUB_ENV
-      - name: Build and push image
-        id: docker-build-push
+      - name: Login to Public Registry
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.AWS_KEY_ID }}
+          password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: Build and push pre-release
         if: >
-          inputs.prerelease
-          && ${{ !github.events.inputs.stable_version }}
-        run: |
-          docker build -t covalent .
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:latest
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:$TAG
-          if [ "${{ github.events.inputs.test_release }}" == false ] ; then
-            docker push -a public.ecr.aws/covalent/covalent
-          fi
-      - name: Update stable tag
+          github.event_name == 'workflow_call'
+          && github.events.inputs.pre_release == 'true'
+        uses: docker/build-push-action@v2
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          context: .
+          file: ./Dockerfile
+          platforms: all
+          push: true
+          tags: |
+            public.ecr.aws/covalent/covalent:latest
+            public.ecr.aws/covalent/covalent:${{ env.TAG }}
+      - name: Tag stable release
         if: >
-          github.event.inputs.stable_version
```

Comment:
we should keep this; we only want to do the stable release if the `stable_version` exists